### ISSUE TITLE
TypedDict diamond inheritance fails with `Overwriting TypedDict field "x" while merging`

### ISSUE THREAD

**Issue by EvanKrall** [python/mypy#8714]
This file:

```
from mypy_extensions import TypedDict


class Grandparent(TypedDict):
    a: str


class ParentB(Grandparent):
    b: str


class ParentC(Grandparent):
    c: str


class Child(ParentB, ParentC):
    pass
```

Produces the error ```test.py:16: error: Overwriting TypedDict field "a" while merging```

I expected this to not error - in some sense, we aren't overwriting field `a` -- it's ultimately coming from the same place through both branches of the inheritance graph.

Maybe more generally, it should not be an error to overwrite a TypedDict field with the exact same type?

**Comment by JukkaL** [comment]
Yeah, if the type is always the same, this message doesn't seem useful.

**Comment by GPHemsley** [comment]
Looks like this is related (though it predates TypedDict): #4354
